.github/workflows/android-test.yaml

```diff
@@ -0,0 +1,39 @@
+name: Test-Android
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+    branches:
+      - '*'
+jobs:
+  android-test:
+    runs-on: ubuntu-latest
+
+    env:
+      FLUTTER_CHANNEL: stable
+      FLUTTER_VERSION: 3.22.2
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - run: echo API_KEY=${{ secrets.TEST_API_KEY }} > example/.env
+
+      - name: Set up Flutter
+        uses: subosito/flutter-action@v2
+        with:
+          flutter-version: ${{ env.FLUTTER_VERSION }}
+          channel: ${{ env.FLUTTER_CHANNEL }}
+
+      - name: Enable KVM
+        run: |
+          echo 'KERNEL=="kvm", GROUP="kvm", MODE="0666", OPTIONS+="static_node=kvm"' | sudo tee /etc/udev/rules.d/99-kvm4all.rules
+          sudo udevadm control --reload-rules
+          sudo udevadm trigger --name-match=kvm
```

Comment:
can we add some comments around why this is needed. Where does it come from etc?